### ISSUE TITLE
Validation Set

### ISSUE THREAD

**Issue by botanyldrm** [yingyichen-cyy/Nested-Co-teaching#2]
Hi, thanks for this great paper and repository. I have a simple question related to code. As I understood, a clean validation set is used during trainings of CIFAR10 and CIFAR100.  Can ratio_val be set to 0 without any problem for trainings? I do not want to use any clean subset for my experiments and just want to see average of test accuracy obtained at the last few epochs.

**Comment by yingyichen-cyy** [comment]
Dear botanyldrm,
Sorry for replying this late and thank you so much for supporting our work! 
You can use the complete training set to train your network and use the average test accuracy obtained at the last few epochs to evaluate, since after applying nested dropout to the network, the network is less prone to overfit on the training set. In other words, network without nested dropout is prone to have increasing test accuracy in the very beginning and then encounter a significant drop in accuracy as it starts to overfit the noise in the training set. While for network with nested dropout applied, even though there is an oscillation w.r.t. test accuracy in the end of training, it would not encounter a significant accuracy drop since it network itself is quite robust and less likely to overfit the noise in the training set. 

An example can be found in the "[valTop1]" for the nested dropout network on CIFAR-100 with 20% label noise in [[cifar100_history.json]](https://drive.google.com/drive/folders/1jsXjmmjsbVmz34Ji_fMzAks-ttP9o9L1). The valTop1 becomes stable at around 57% in the last few training epochs.
We use a clean validation set in our paper only to achieve even better results.

**Comment by botanyldrm** [comment]
Thanks a lot for your answer yingyichen-cyy.
